index.html

```diff
@@ -101,19 +199,623 @@ <h3>
+          {{WritableStreamDefaultWriter}} can be created and released as necessary.
+          This example uses a `TextEncoder` to encode a {{DOMString}} as the
+          necessary {{Uint8Array}} for transmission.
+
+          <pre class="js">
+            const encoder = new TextEncoder();
+
+            const { writable } = await socket.connection;
+            const writer = writable.getWriter();
+
+            await writer.write(encoder.encode("PING"));  
+
+            writer.releaseLock();
+          </pre>
+
+          The {{WritableStreamDefaultWriter/write()}} method returns a {{Promise}} which
+          resolves when data has been written. While having some data available in
+          the transmit buffer is important to maintain good throughput awaiting this
+          {{Promise}} before generating too many chunks of data is a good practice
+          to avoid excessive buffering.
+        </aside>
+
+        The steps to <dfn data-lt="initialize tcp writable stream">initialize the TCP writable stream</dfn> are:
+
+        <ol>
+          <li>Let |stream:WritableStream| be a [=new=] {{WritableStream}}.
+          <li>Let |signal:AbortSignal| be |stream|'s [=WritableStream/signal=].
+          <li>
+            Let |writeAlgorithm| be the following steps, given |chunk|:
+            <ol>
+              <li>Let |promise:Promise| be [=a new promise=].
+              <li>Assert: |signal| is not [=AbortSignal/aborted=].
+              <li>
+                If |chunk| cannot be [=converted to an IDL value=] of type
+                {{BufferSource}}, reject |promise| with a {{TypeError}} and return
+                |promise|. Otherwise, save the result of the conversion in
+                |source:BufferSource|.
+              <li>
+                <li>
+                  [=In parallel=], run the following steps:
+                  <ol>
+                    <li>
+                      Invoke the operating system to write |bytes| to the socket.
+
+                      <div class="note">
+                        The operating system may return from this operation once
+                        |bytes| has been queued for transmission rather than after it
+                        has been transmitted.
+                      </div>
+                    <li>
+                      [=Queue a global task=] on the [=relevant global object=] of
+                      [=this=] using the [=TCP socket task source=] to run the
+                      following steps:
+                      <ol>
+                        <li>
+                          If the chunk was successfully written, [=resolve=] |promise| with `undefined`.
+
+                          <div class="note">
+                            [[STREAMS]] specifies that |writeAlgorithm| will only be
+                            invoked after the {{Promise}} returned by a previous
+                            invocation of this algorithm has resolved.
```

Comment:
You're missing the "for efficiency" part of this note that's in the Web Serial API specification.

Reply:
Fixed.